src/LoopInterface.php

```diff
@@ -240,6 +240,34 @@ public function isTimerActive(TimerInterface $timer);
     public function futureTick(callable $listener);
 
     /**
+     * Registers a signal listener with the loop, which
+     * on it's turn registers it with a signal handler
+     * suitable for the loop implementation.
+     *
+     * See also [example #4](examples).
+     *
+     * @param int $signal
+     * @param callable $listener
+     *
+     * @return void
+     */
+    public function onSignal($signal, callable $listener);
+
+    /**
+     * Removed previous registered signal listener from
+     * the loop, which on it's turn removes it from the
+     * underlying signal handler.
+     *
+     * See also [example #4](examples).
+     *
+     * @param int $signal
+     * @param callable $listener
+     *
+     * @return void
+     */
+    public function removeSignal($signal, callable $listener);
```

Comment:
Can the same listener be added multiple times? If so, which one will be removed here?

Also, the docs should probably include what happens if the given signal or listener is not attached (NO-OP just like other methods?).